Review and Integrate open-1167, entanglement actions

larkin opened the pull request:
The entanglement bundle defines move, copy, and link actions, and
exposes them as context menu actions.

* The Move action moves an object from it's current parent to a new
parent object.
* The Copy action deep-copies an object to a new parent object.
* The Link action links an object to a new parent object.

These actions are implemented by three new services: moveService,
copyService, and linkService.

Mocks are provided for each service for easy testing of components
that depend on them.

Additionally, this bundle provides a DomainObjectFactory that simplifies
the construction of mockDomainObjects for tests.

These actions are exposed to the user as context menu options.

# Author Checklist
1. Changes address original issue? Yes, ish.  The original issue is an umbrella for a number of changes.
2. Unit tests included and/or updated with changes? Yes
3. Command line build passes? Yes
4. Expect to pass code review? No
5. Project-specific information isolated to appropriate branches? Yes

@VWoeltjen let me know if you have any questions.

VWoeltjen commented on platform/entanglement/test/suite.json:
```diff
@@ -0,0 +1,9 @@
+    "services/LinkService",
+    "services/MoveService",
+    "services/LocationService"
+]
```
Missing newline here

VWoeltjen commented on platform/entanglement/src/actions/CopyAction.js:
```diff
@@ -0,0 +1,63 @@
+    function () {
+        "use strict";
+
+        function CopyAction(locationService, copyService, context) {
```
Would like some class-level JSDoc here and in other scripts

VWoeltjen commented on platform/entanglement/src/actions/CopyAction.js:
```diff
@@ -0,0 +1,63 @@
+/*global define */
```
It's a recent change, but we should have open source license headers on these

VWoeltjen commented:
Code changes look good, see notes above for housekeeping nits

As discussed, a little concerned that Move is allowed for containers that cannot be modified, but makes sense to keep scope sane (since this is in a separate issue.) Looking at the code, I see Move follows up with a remove, which I expect will already be prohibited, so I'd expect Move to end up functioning like Link in this case (not desired behavior, but not really a regression either)

larkin commented:
@VWoeltjen Ok, will take care of housekeeping.  When I have completed that, should I ask you for another review or should I go ahead and merge?

VWoeltjen commented:
Also, let's pick some more reasonable glyphs:

![image](https://cloud.githubusercontent.com/assets/1685764/8119485/a6c05a98-104a-11e5-960f-054e0824818b.png)

See UI Symbols Font Map in the Google Drive (looks like Move/Copy/Link are unicode 00F3 00F2 00F4  respectively)

Go ahead and ping me for another review when these changes are in so I can complete the checklist.

larkin commented:
@VWoeltjen Ok, I've addressed the housecleaning and fixed the bug we found yesterday where specific objects couldn't be moved/copied/linked.

I updated the glyphs with the unicode points for those icons but they appear to be missing in my branch:
![screen shot 2015-06-12 at 9 54 13 am](https://cloud.githubusercontent.com/assets/226503/8135285/03edf040-10eb-11e5-96a3-870366066497.png)
Since the glyphs were already preliminary, I wouldn't consider this blocking for merge-- I'll re-open the glyph request issue.

Please proceed with review and integration.

VWoeltjen commented:
Agreed, don't want to block this waiting for the correct glyphs. I'll swap in some reasonable-looking ones when I complete the merge

![screen shot 2015-06-12 at 10 23 25 am](https://cloud.githubusercontent.com/assets/1685764/8135621/7481cc12-10ed-11e5-83ce-b735c0f83278.png)

VWoeltjen commented:
Review Checklist

1. Changes appear to address issue? Y (w/i scope as described in pull request)
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
5. Project-specific information isolated to appropriate branches? Y

Integration Checklist

1. Automated build passing after merge? Y
2. No merge conflicts (or resolution trivial/obvious)? Y
3. All master branches up-to-date after merge? Y
4. Project-specific information isolated to appropriate branches? Y
